app/controllers/identity/email_verifications_controller.rb

```diff
@@ -1,28 +1,19 @@
 class Identity::EmailVerificationsController < ApplicationController
   skip_before_action :authenticate_user!, only: :show
 
-  before_action :set_user, only: :show
-
   def show
-    @user.update! verified: true
-    redirect_to root_path, notice: "Thank you for verifying your email address"
+    verification = User::EmailVerification.find_by_token(params[:sid])
+
+    if verification
+      verification.verify!
+      redirect_to root_path, notice: "Thank you for verifying your email address"
+    else
+      redirect_to edit_identity_email_path, alert: "That email verification link is invalid"
```

Comment:
Far as I can tell we omit sign in from `show` but then here we redirect to `identity/emails#show` which requires sign in. So this seems borked.